File: searching/binarySearch.cpp
#include <bits/stdc++.h>
using namespace std;

int binarySearch(int number, int size, int arr[])
{
  int start = 0;
  int end = size - 1;
  int mid = start + (end - start) / 2;
  while (start <= end)
  {
    if (arr[mid] == number)
    {
      return mid;
    }
    if (number > arr[mid])
    {
      start = mid + 1;
    }
    else
    {
      end = mid - 1;
    }
    mid = start + (end - start) / 2; 
  }
  return -1; // Not found
}

int main()
{
  int even[6] = {1, 2, 3, 4, 5, 6};
  int odd[5] = {12, 23, 45, 78, 98};
  cout << "Index of 5 in even array: " << binarySearch(5, 6, even) << endl;
  cout << "Index of 23 in odd array: " << binarySearch(23, 5, odd) << endl;
  return 0;
}

File: bootcamp/week-6-searching-sorting/binary-search.py
arr = [1, 2, 3, 4, 5]
num = int(input("Enter a number to search: "))
def binary_search(arr, num):
  low = 0
  high = len(arr) - 1
  while low <= high:
    mid = (low + high) // 2
    if arr[mid] == num:
      return mid
    elif arr[mid] < num:
      low = mid + 1
    else:
      high = mid - 1
  return -1
  
result = binary_search(arr, num)
if result != -1:
  print(f"Number found at index: {result}")
else:
  print("Number not found in the array.")
